packages/hash/frontend/src/blocks/page/Comments/CommentTextField.tsx

```diff
@@ -0,0 +1,194 @@
+import { mentionNodeView } from "../MentionView/MentionNodeView";
+
+type CommentTextFieldProps = {
+  blockId: string;
+  onClose: () => void;
+  onSubmit: ((content: TextToken[]) => Promise<void>) | null;
+};
+
+export const CommentTextField: FunctionComponent<CommentTextFieldProps> = ({
+  blockId,
+  onClose,
+  onSubmit,
+}) => {
+  const viewRef = useRef<EditorView<Schema>>();
+  const [portals, renderPortal] = usePortals();
+  const { accountId } = useRouteAccountInfo();
+  const [loading, setLoading] = useState(false);
+
+  const [prevBlockId, setPrevBlockId] = useState("");
+  const [editorContainer, setEditorContainer] = useState<HTMLDivElement | null>(
+    null,
+  );
+
+  const createEditor = useCallback(
+    (container: HTMLDivElement) => {
+      const schema = createSchema({
+        doc: {
+          content: "inline*",
+        },
+        text: textNode,
+        hardBreak: hardBreakNode,
+        mention: mentionNode,
+      });
+
+      const doc = schema.node("doc", {}, []);
+
+      const state = EditorState.create<Schema>({
+        doc,
+        plugins: [
+          keymap<Schema>(baseKeymap),
+          ...createFormatPlugins(renderPortal),
+          formatKeymap(doc),
+          createSuggester(renderPortal, accountId, container),
+          placeholderPlugin(renderPortal, "Leave a comment"),
+        ],
+      });
+
+      const view = new EditorView<Schema>(container, {
+        state,
+        clipboardTextSerializer: clipboardTextSerializer(
+          state.schema.nodes.hardBreak,
+        ),
+        nodeViews: {
+          mention: mentionNodeView(renderPortal, accountId),
+        },
+        handleKeyDown: (_, { shiftKey, key }) => {
+          if (!shiftKey && key === "Enter") {
+            if (onSubmit && viewRef.current?.state.doc.content) {
+              const tokens =
+                viewRef.current.state.doc.content.toJSON() as TextToken[];
+
+              setLoading(true);
+              onSubmit(tokens)
+                .then(() => {
+                  onClose();
+                })
+                .finally(() => {
+                  setLoading(false);
+                });
+            }
+
+            return true;
+          }
+
+          return false;
+        },
+      });
+
+      view.dom.classList.add(styles.Prosemirror_Input!);
+
+      view.focus();
+
+      viewRef.current = view;
+    },
+    [accountId, onSubmit, renderPortal, onClose],
+  );
+
+  if (editorContainer && prevBlockId !== blockId) {
+    setPrevBlockId(blockId);
+
+    editorContainer.innerHTML = "";
+    createEditor(editorContainer);
+  }
+
+  return (
+    <Box
+      sx={({ transitions, palette }) => ({
+        width: 250,
+        display: "flex",
+        borderRadius: 1.5,
+        border: `1px solid ${palette.gray[30]}`,
+        backdropFilter: "blur(40px)",
+        transition: transitions.create("border-color"),
+        "&:focus-within": {
+          borderColor: palette.blue[60],
+        },
```

Comment:
I suggest using `styled` from `@mui/material` to increase readability by keeping the `jsx` short & clean.
Same for all the long inline `sx` definitions on this file. Here is an example which uses `experimental_sx` as well https://github.com/hashintel/hash/blob/d3ff11efa4f84a384200ed64d8c431085ee1f817/packages/blocks/shuffle/src/components/add-entities-dialog.tsx#L20-L30

Reply:
I guess this is subjective but my opinion is that this doesn't really increase the readibility of the jsx. Sure, you skip through the styles of this element by placing them somewhere else, but if you want to focus on the styles you are actually decreasing readability by having to jump through an extra hoop to get there. This is a great solution whenever we want to reuse a component with the same styles in multiple places but don't really want to create a react component for it, but I don't think that applies here.